tests/subsys/pm/device_runtime_api/src/main.c

```diff
@@ -190,6 +191,35 @@ static void test_api(void)
 
 	(void)pm_device_state_get(dev, &state);
 	zassert_equal(state, PM_DEVICE_STATE_ACTIVE, NULL);
+
+	/* Test state lock API */
+	pm_device_state_lock(dev);
+
+	/* The device should not be locked because device runtime is
+	 * enabled
+	 */
+	locked = pm_device_state_is_locked(dev);
+	zassert_equal(locked, false, NULL);
+
+	(void)pm_device_state_get(dev, &state);
+	zassert_equal(state, PM_DEVICE_STATE_ACTIVE, NULL);
+
+	ret = pm_device_runtime_disable(dev);
+	zassert_equal(ret, 0, NULL);
+
+	pm_device_state_lock(dev);
+
+	locked = pm_device_state_is_locked(dev);
+	zassert_equal(locked, true, NULL);
+
+	/* Enable device runtime should fail because of the state lock */
+	pm_device_runtime_enable(dev);
```

Comment:
I believe this is a fallible function in @gmarull's pr #40670 which this could benefit from perhaps